spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/DefaultSource.scala

```diff
@@ -0,0 +1,93 @@
+  override def createRelation(sqlContext: SQLContext,
+                              mode: SaveMode,
+                              parameters: Map[String, String],
+                              data: DataFrame): BaseRelation = {
+    val table = sqlContext.getDataSourceRelation(getTable(parameters), getKeyspace(parameters),
+      getScanType(parameters), getCluster(parameters), None)
+
+    mode match {
+      case Append => table.insert(data, false)
+      case Overwrite => throw new UnsupportedOperationException("Overwriting a Cassandra Table is not allowed.")
+      case ErrorIfExists =>
+        if (!table.buildScan().isEmpty()) {
+          throw new UnsupportedOperationException("Writing to a none-empty Cassandra Table is not allowed.")
+        } else {
+          table.insert(data, false)
+        }
+      case Ignore =>
+    }
+
+    sqlContext.getDataSourceRelation(getTable(parameters), getKeyspace(parameters),
+      getScanType(parameters), getCluster(parameters), None)
+  }
+
+
+  import DefaultSource._
+
+  private def getTable(parameters: Map[String, String]) : String =
+    parameters.getOrElse(CassandraDataSourceTableNameProperty, missingProp(CassandraDataSourceTableNameProperty))
+
+  private def getKeyspace(parameters: Map[String, String]) : String =
+    parameters.getOrElse(CassandraDataSourceKeyspaceNameProperty, missingProp(CassandraDataSourceKeyspaceNameProperty))
+
+  private def getCluster(parameters: Map[String, String]): Option[String] =
+    parameters.get(CassandraDataSourceClusterNameProperty)
+
+  private def getUserDefinedSchema(parameters: Map[String, String]) : Option[String] =
+    parameters.get(CassandraDataSourUserDefinedSchemaNameProperty)
+
+  private def missingProp(prop: String) = throw new IllegalArgumentException(s"Missing $prop name")
+
+  private def getScanType(parameters: Map[String, String]) : ScanType = CassandraDataSourceScanTypeMap.get(
+    parameters.getOrElse(CassandraDataSourceScanTypeNameProperty,
+      CassandraDataSourcePrunedFilteredScanTypeName).toLowerCase).getOrElse(CatalystScanType)
+}
+
+object DefaultSource {
+  val CassandraDataSourceTableNameProperty = "c_table"
+  val CassandraDataSourceKeyspaceNameProperty = "keyspace"
+  val CassandraDataSourceClusterNameProperty = "cluster"
+  val CassandraDataSourceScanTypeNameProperty = "scan_type"
+  val CassandraDataSourUserDefinedSchemaNameProperty = "schema"
+
+  val CassandraDataSourceBaseScanTypeName = "base"
+  val CassandraDataSourcePrunedScanTypeName = "pruned"
+  val CassandraDataSourcePrunedFilteredScanTypeName = "pruned_filtered"
+  val CassandraDataSourceCatalystScanTypeName = "catalyst"
+  val CassandraDataSourceScanTypeMap = Map (
+    CassandraDataSourceBaseScanTypeName -> BaseScanType,
+    CassandraDataSourcePrunedScanTypeName -> PrunedScanType,
+    CassandraDataSourcePrunedFilteredScanTypeName -> PrunedFilteredScanType,
+    CassandraDataSourceCatalystScanTypeName -> CatalystScanType
+  )
```

Comment:
Hint: it seems we're always failing over to a safe default value `CatalystScanType`. Maybe consider using a default value for the map:

```scala
  val CassandraDataSourceScanTypeMap = Map(
    CassandraDataSourceBaseScanTypeName -> BaseScanType,
    CassandraDataSourcePrunedScanTypeName -> PrunedScanType,
    CassandraDataSourcePrunedFilteredScanTypeName -> PrunedFilteredScanType
  ).withDefaultValue(CatalystScanType)
```